nilearn/plotting/find_cuts.py

```diff
@@ -53,66 +54,72 @@ def find_xyz_cut_coords(img, mask=None, activation_threshold=None):
     img = check_niimg_3d(img)
     data = _safe_get_data(img)
 
-    # To speed up computations, we work with partial views of the array,
-    # and keep track of the offset
-    offset = np.zeros(3)
+    # We have 3 potential masks here:
+    # - a mask provided by the user
+    # - the activation threshold
+    # - if data is a masked_array, the associated mask
+
+    masks = []
+    if mask is not None:
+        masks.append[mask]
+
+    # Account for numerical noise
+    if activation_threshold is None:
+        activation_threshold = 1e-7
+
+    # If threshold is auto, it is computed later on masked data
+    if activation_threshold is not None and activation_threshold != 'auto':
+        masks.append(new_img_like(img, np.abs(data) >= activation_threshold))
 
-    # Deal with masked arrays:
     if hasattr(data, 'mask'):
-        not_mask = np.logical_not(data.mask)
-        if mask is None:
-            mask = not_mask
-        else:
-            mask *= not_mask
-        data = np.asarray(data)
+        masks.append(new_img_like(np.logical_not(data.mask.astype(np.int8))))
 
-    # Get rid of potential memmapping
-    data = as_ndarray(data)
-    my_map = data.copy()
-    if mask is not None:
-        # check against empty mask
-        if mask.sum() == 0.:
-            warnings.warn(
-                "Provided mask is empty. Returning center of mass instead.")
-            cut_coords = ndimage.center_of_mass(np.abs(my_map)) + offset
-            x_map, y_map, z_map = cut_coords
-            return np.asarray(coord_transform(x_map, y_map, z_map,
-                                              img.get_affine())).tolist()
-        slice_x, slice_y, slice_z = ndimage.find_objects(mask)[0]
-        my_map = my_map[slice_x, slice_y, slice_z]
-        mask = mask[slice_x, slice_y, slice_z]
-        my_map *= mask
-        offset += [slice_x.start, slice_y.start, slice_z.start]
+    if len(masks) > 0:
+        mask = intersect_masks(masks, threshold=1)
+        data = unmask(apply_mask([img], mask)[0], mask)
+    else:
+        # Get rid of potential memmapping
+        data = as_ndarray(data)
+        # XXX Is that necessary given _saf_get_data?
+        data = data.copy()
```

Comment:
I believe that the copy is necessary to get rid of the memmapping in a robust way. However, I am not sure how useful this is.